engine/src/deploy.cpp

```diff
@@ -89,6 +89,57 @@ extern Boolean InitSSLCrypt(void);
 
 ////////////////////////////////////////////////////////////////////////////////
 
+static const char *kMCDeployArchitectureStrings[] =
+{
+    "",
+    "i386",
+    "x86-64",
+    "armv6",
+    "armv7",
+    "armv7s",
+    "arm64",
+    "ppc",
+    "ppc64",
+    nil,
+};
+
+static bool MCDeployMapArchitectureString(const MCString& p_string, MCDeployArchitecture& r_architecture)
+{
+    for(uindex_t i = 0; kMCDeployArchitectureStrings[i] != nil; i++)
+    {
+        // As 'p_string' is an MCString the '==' operator does a caseless comparison.
+        if (p_string == kMCDeployArchitectureStrings[i])
+        {
+            r_architecture = (MCDeployArchitecture)i;
+            return true;
+        }
+    }
+    
+    return false;
+}
+
+static Exec_stat MCDeployPushMinOSVersion(MCDeployParameters& p_params, MCDeployArchitecture p_arch, const char *p_vers_string)
+{
+    int t_major, t_minor, t_inc;
+    t_major = t_minor = t_inc = 0;
+    sscanf(p_vers_string, "%d.%d.%d", &t_major, &t_minor, &t_inc);
```

Comment:
I'm assuming that you don't check the return value of `sscanf()` because if it returns < 3 then you just assume that some components of the version string were missing.  They get initialised anyway, after all! :wink: